experimental/ast/decl.go

```diff
@@ -23,7 +23,7 @@ import (
 )
 
 const (
-	DeclKindNil DeclKind = iota
+	DeclKindZero DeclKind = iota
```

Comment:
Maybe we could call the zero _kinds_ "Invalid" or "Undefined"? So a zero value has an invalid/undefined kind. WDYT?

Reply:
Sure, invalid is fine.

Reply:
This comment was really directed at _all_ kinds, not just `DeclKind`. Sorry if that wasn't clear. See #405.